Tools/PlatformWPE.cmake

```
@@ -85,6 +100,7 @@ if (ENABLE_COG)
             -Dwpe_api=${WPE_API_VERSION}
             -Dplatforms=${WPE_COG_PLATFORMS}
             -Db_sanitize=${COG_MESON_SANITIZE_OPTION}
+            -Dlibportal=${COG_MESON_LIBPORTAL_OPTION}
```

You can change this one like to pass `-Dlibportal=auto` and let Meson detect whether libportal is available, and then you can remove the `FINDLIBPORTAL.cmake` file and all the rest of changes from this PR.

```suggestion
            -Dlibportal=auto
```

Really, this should be the only needed change 😉

---

Of course, keep the GTK 4 check, and add `gtk4` to the list of platforms only if present. My comment above is only regarding the `libportal` detection.